sycl/doc/extensions/experimental/sycl_ext_oneapi_cuda_async_barrier.asciidoc

```diff
@@ -0,0 +1,390 @@
+call corresponding to this arrival. This must not be the last arrival thet 
+causes the cycle to complete - it would be undefined behavior. That means 
+`count` must be strictly lower than the remaining number of arrivals required 
+to complete this cycle.
+
+This can be used to signal many arrivals by one function call. However, it 
+should not be used on its own, as it can not be the last arrival in a cycle. 
+So it should either be followed with a call to `arrive` or by a barrier 
+operation, such as (such as `group_barrier()`) after which a different 
+work-item is guaranteed to call `arrive` within the same cycle. 
+
+`void arrive_copy_async()` : Schedules arrive operation to be triggered
+asynchronously when all previous asynchronous memory copies initiated by the 
+calling work item complete. Before the arrive operation is triggered, the 
+pending count on the barrier is increased by 1, so after the arrival there is 
+no change to the pending count. Pending count with the increase by this call 
+must not exceed the value returned by `max`. If it does, it causes undefined 
+behavior.
+
+`void arrive_copy_async_no_inc()` : Schedules arrive operation to be triggered 
+asynchronously when all previous asynchronous memory copies initiated by the 
+calling work item complete.
+
+`void wait(arrival_token arrival)` : Executes wait operation, blocking until 
+the predetermined number of work items have called `arrive`.
+
+`bool test_wait(arrival_token arrival)` : Checks whether all the arrivals have 
+already happened for the current cycle, returning `true` if they did and 
+`false` if `wait(arrival)` would block.
+
+`void arrive_and_wait()` : Equivalent to calling `wait(arrive())`.
+
+`static constexpr uint64_t max()` : Returns the maximum value of the expected 
+and pending counts supported by the implementation.
+
+==== Sample Header
+
+[source, c++]
+----
+namespace sycl::ext::oneapi::experimental::cuda {
+
+class barrier {
+  [implementation defined internal state]
+
+public:
+  using arrival_token = [implementation defined];
+
+  // barriers cannot be moved or copied
+  barrier(const barrier &other) = delete;
+  barrier(barrier &&other) noexcept = delete;
+  barrier &operator=(const barrier &other) = delete;
+  barrier &operator=(barrier &&other) noexcept = delete;
+
+  void initialize(uint32_t expected_count);
+  void invalidate();
+  arrival_token arrive();
+  arrival_token arrive_and_drop();
+  arrival_token arrive_no_complete(int32_t count);
+  arrival_token arrive_and_drop_no_complete(int32_t count);
+  void arrive_copy_async();
+  void arrive_copy_async_no_inc();
+  void wait(arrival_token arrival);
+  bool test_wait(arrival_token arrival);
+  void arrive_and_wait();
+  static constexpr uint64_t max();
+};
+
+} // namespace sycl::ext::oneapi::experimental::cuda
+----
+
+== Examples
+
+[source, c++]
+----
+using namespace sycl;
+using namespace sycl::ext::oneapi::cuda;
```

Comment:
Just a small nit:

```suggestion
using namespace sycl::ext::oneapi::experimental::cuda;
```

Same for the other examples below.

Reply:
done